### PR TITLE
Compile with Nim 0.20

### PR DESCRIPTION
**Pull request by kidandcat**
This PR is only to bring NimForum to Nim 0.20 + latest Karax

It should be reviewed because, for example, just MD5 a random string is not the best approach to get a salt. Bcrypt was removed (it doesn't compile, at least in windows).

### CODE REVIEW

File: src/frontend/newthread.nim [hunk 1]
```diff
@@ -36,33 +36,33 @@ when defined(js):
...
+      section(class = "container grid-xl"):
+        tdiv(id = "new-thread"):
+          tdiv(class = "title"):
             p(): text "New Thread"
-          tdiv(class="content"):
-            input(id="thread-title", class="form-input", `type`="text", name="subject",
-                  placeholder="Type the title here",
-                  oninput=(e: Event, n: VNode) => onSubjectChange(e, n, state))
+          tdiv(class = "content"):
+            input(id = "thread-title", class = "form-input", `type` = "text", name = "subject",
```
Comment (dom96) [inline src/frontend/newthread.nim]: Are all these formatting changes required for 0.20.0?

Comment (kidandcat) [reply src/frontend/newthread.nim]: Nop, I'll revert them, vscode + nim extension formated it without me noticing

**Comment by kidandcat** [comment]
Isn't any other encryption package available? I don't like to see something awesome like Nimforum to be not crossplatform just because a package used in 3 Loc

**Comment by kidandcat** [comment]
I asnwer myself, bcrypt is Linux only, other implementations for hashing would invalidate your actual Database. Sorry but I will not work on a non cross-platform software

**Comment by dom96** [comment]
> Sorry but I will not work on a non cross-platform software

You could look into creating a bcrypt wrapper/library that is cross-platform. Not working on the Nimforum just because it doesn't support Windows fully is quite disappointing.

**Comment by kidandcat** [comment]
Im thinking about bcrypt, if I don't find a way to go, I'll move to other
Nim area. Anyway I'll spend my time and efforts towards Nim development
because I think it's the definitive language and I have all my hopes
deposited on you all :)

**Comment by dom96** [comment]
If you create a cross-platform bcrypt package you would help out a lot of Nim devs :)

**Pull request closed without merging**